src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs

```diff
@@ -157,12 +157,18 @@ public override bool Execute()
 
                 retValue = BuildEngine.BuildProjectFile(tempProj, new string[] { CompileTargetName }, globalProperties, null);
 
-                // Delete the temporary project file unless diagnostic mode has been requested
+                // Delete the temporary project file and generated files unless diagnostic mode has been requested
                 if (!GenerateTemporaryTargetAssemblyDebuggingInformation)
                 {
                     try
                     {
                         File.Delete(tempProj);
+
+                        DirectoryInfo intermediateOutputPath = new DirectoryInfo(IntermediateOutputPath);
```

Comment:
Can you help me understand the assumptions being made here? 

Are you assuming that the destination where build intermediate artifacts go will be something like `$(IntermediateOutputPath)$(MSBuildProjectName)` ?

Reply:
The destination where intermediate files are placed during compilation for the temporary assembly is 'IntermediateOutputPath'.  'IntermediateOutputPath' is passed in to the BuildEngine through the parameter 'globalProperties'.   Note that the temporary project is placed in a different directory, next to the real project file for the application.